app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,40 @@
+require_relative "../views/groups/render_json"
+
+class GroupsController < ApplicationController
+  include GroupsJson
+
+  def create
+    user = @current_user
+    group = Group.new(name: params.fetch(:group, {})[:name])
+    if group.save
+      group.add_user(user, accepted: true)
+      render status: 201,
+        json: create_success(group)
+    else
+      render status: 404,
+        json: group_errors(group)
+    end
+  end
+
+  def show
```

Comment:
ok, @lpil ths is where a user creates a group. So it doesnt have an existing ID or slug because it doesnt exist yet. so it doesnt make sense to me to look it up by ID or slug. This is why I have created the to_slu method which happens before_save to the group name. The the slug and id are returned to the user. I don't understand how looking up a group that doesnt exist yet by id or slug would work?